services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCAuthorizationDetailsProcessor.java

```diff
@@ -81,9 +88,77 @@ public List<AuthorizationDetailsResponse> process(UserSessionModel userSession,
             throw getInvalidRequestException("no valid authorization details found");
         }
 
+        // For authorization code flow, create CredentialOfferState if credential identifiers are present
+        // This allows credential requests with credential_identifier to find the associated offer state
+        createOfferStateForAuthorizationCodeFlow(userSession, clientSessionCtx, authDetailsResponse);
+
         return authDetailsResponse;
     }
 
+    /**
+     * Creates CredentialOfferState for authorization code flow when credential identifiers are generated.
+     * This is only done for authorization code flow (not pre-authorized flow which already has an offer state).
+     * Processes all OID4VC authorization details to support multiple credential requests.
+     */
+    private void createOfferStateForAuthorizationCodeFlow(UserSessionModel userSession, ClientSessionContext clientSessionCtx,
+                                                          List<AuthorizationDetailsResponse> authDetailsResponse) {
+        AuthenticatedClientSessionModel clientSession = clientSessionCtx.getClientSession();
+        ClientModel client = clientSession != null ? clientSession.getClient() : null;
+        UserModel user = userSession != null ? userSession.getUser() : null;
+
+        if (client == null || user == null) {
+            return;
+        }
+
+        // Skip if we're in pre-authorized code flow (it already has an offer state that will be updated)
+        // Pre-authorized flow sets VC_ISSUANCE_FLOW note on the client session
+        if (clientSession != null) {
```

Comment:
This condition is always true.
In Line 106, `client == null` if `clientSession == null`.
In Line 109, the condition is true, so we does not reach Line 115.